x-pack/plugins/maps/public/kibana_services.ts

```diff
@@ -5,17 +5,25 @@
  * 2.0.
  */
 
+import React from 'react';
 import type { CoreStart } from 'kibana/public';
 import type { MapsEmsConfig } from '../../../../src/plugins/maps_ems/public';
 import type { MapsConfigType } from '../config';
-import type { MapsPluginStartDependencies } from './plugin';
+import type { MapsPluginSetupDependencies, MapsPluginStartDependencies } from './plugin';
 import type { EMSSettings } from '../common/ems_settings';
 import type { PaletteRegistry } from '../../../../src/plugins/charts/public';
 
 let kibanaVersion: string;
 export const setKibanaVersion = (version: string) => (kibanaVersion = version);
 export const getKibanaVersion = () => kibanaVersion;
 
+let pluginsSetup: MapsPluginSetupDependencies;
```

Comment:
setup plugins are supposed to only be used during setup phase and should not be used afterwards. To avoid a pattern where its easy to consume setup dependencies, how about not putting setup in Kibana services and instead passing ApplicationUsageTrackingProvider directly to renderApp since its called from setup?

Reply:
> setup plugins are supposed to only be used during setup phase and should not be used afterwards.

App `mount` related operations blur the lines on this given this is a setup plugin component that is meant to be leveraged in a start context.

tbh I was split on this. The reason I didn't pass it through `renderApp` is that it seems junky to have `params: AppMountParameters`, which is this universal core type that every app uses in the plugin `mount` function and seems like it should cover everything + some extra react component passed from a file that isn't doing anything with react. That said, it's not hard to change if there are strong feelings around this.

Reply:
I think its worth changing to avoid exposing setup via kibana_services. I think adding another argument to renderApp better contains setup dependencies.

Reply:
Ok. Also discussed offline with @thomasneirynck and he agrees with the change. It's updated.